src/Modal.js

```diff
@@ -161,7 +161,11 @@ const Modal = React.createClass({
     let show = !!props.show;
     let dialog = React.Children.only(this.props.children);
 
-    let setMountNode = ref => this.mountNode = (!ref || ref.getMountNode());
+    let setMountNode = ref => {
+      if (ref !== null && typeof ref === 'object') {
```

Comment:
this really just needs to be: `this.mountNode = (ref ? ref.getMountNode() : ref)` otherwise the mountPoint won't actually get nulled out correctly